BinaryEncoding.md

```diff
@@ -240,6 +239,42 @@ This section must be preceded by a [Functions](#functions-section) section.
 | count | `varuint32` | count of entries to follow |
 | entries | `uint16*` | repeated indexes into the function table |
 
+### Function Names section
+
+ID: `function_names`
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| names | `utf8-C-string*` | sequence of null-terminated utf8-encoded strings |
```

Comment:
+1 for up-front length and no terminator. -1 for disallowing internal 0 -- an embedder may impose additional constraints like that, but I don't see why the Wasm binary format needs to.

For what's it worth, even JavaScript can deal with 0 characters in the middle of strings just fine:
```
let s = 'a\0b'
s.charCodeAt(1) === 0  // true
let o = {[s]: 7}
o[s] === 7  // true
```